packages/esm-patient-vitals-app/src/vitals-biometrics-form/vitals-biometrics-form.utils.ts

```diff
@@ -1,7 +1,19 @@
 import isNumber from 'lodash/isNumber';
 import { type ConceptMetadata } from '../common';
 
-export function calculateBodyMassIndex(weight: number, height: number): number {
+export function calculateBodyMassIndex(weight: number, height: number, weightUnit: string, heightUnit: string): number {
+  if (weightUnit == 'lb' || weightUnit == 'lbs') {
+    weight = weight * 0.45359237;
+  }
+  if (weightUnit == 'g') {
+    weight = weight / 1000;
+  }
+  if (heightUnit == 'm') {
+    height = height * 100;
+  }
+  if (heightUnit == 'in') {
+    height = height * 0.0254;
+  }
   if (!weight || !height) return;
```

Comment:
This should still be the first line. No point in trying to "convert" an undefined height or weight. We should probably also do the conversion after the `> 0` guards.